src/dotty/tools/dotc/core/Names.scala

```diff
@@ -347,4 +347,27 @@ object Names {
       StringBuilder.newBuilder.mapResult(s => from.fromChars(s.toCharArray, 0, s.length))
     def apply(): Builder[Char, Name] = termNameBuilder
   }
+
+  implicit val NameOrdering: Ordering[Name] = new Ordering[Name] {
+    def compare(x: Name, y: Name): Int = {
+      if (x.isTermName && y.isTypeName) 1
+      else if (x.isTypeName && y.isTermName) -1
+      else if (x.start == y.start && x.length == y.length) 0
+      else {
+        val until = Math.min(x.length, y.length)
```

Comment:
I believe at some moment we had a meeting where we agreed to have a style guideline to minimize usage of infix notation. Didn't we?